src/controllers/organizationController.ts

```diff
@@ -414,6 +383,86 @@ export default class OrganizationController {
     return none;
   }
 
+  private createOnboardingDocuments(
+    ipaCode: string,
+    requestsCreationResponseSuccess: IResponseSuccessCreation<RequestCollection>
+  ): TaskEither<
+    IResponseErrorInternal,
+    IResponseSuccessCreation<RequestCollection>
+  > {
+    // TODO:
+    //  the documents must be stored on cloud (Azure Blob Storage).
+    //  @see https://www.pivotaltracker.com/story/show/169644958
+    const outputFolder = `./documents/${ipaCode}`;
+    const createDocumentPerRequest = (request: Request) => {
+      const rejectError = (maybeError: Option<Error>) => {
+        if (isSome(maybeError)) {
+          return Promise.reject(maybeError.value);
+        }
+      };
+      return tryCatch<IResponseErrorInternal, undefined>(
+        () => {
+          if (OrganizationRegistrationRequest.is(request)) {
+            return this.documentService
+              .generateDocument(
+                localeIt.organizationController.registerOrganization.contract.replace(
+                  "%s",
+                  `${request.organization.ipa_code} ${request.organization.fiscal_code}`
+                ),
+                `${outputFolder}/${request.document_id}.pdf`
+              )
+              .then(rejectError);
+          }
+          if (UserDelegationRequest.is(request)) {
+            return this.documentService
+              .generateDocument(
+                // TODO:
+                //  refactor this operation using an internationalization framework allowing params interpolation in strings.
+                //  @see https://www.pivotaltracker.com/story/show/169644146
+                localeIt.organizationController.registerOrganization.delegation
+                  .replace(
+                    "%legalRepresentative%",
+                    `${request.organization.legal_representative.given_name} ${request.organization.legal_representative.family_name}`
+                  )
+                  .replace("%organizationName%", request.organization.name)
+                  .replace(
+                    "%delegate%",
+                    `${request.requester.given_name} ${request.requester.family_name}`
+                  ),
+                `${outputFolder}/${request.document_id}.pdf`
+              )
+              .then(rejectError);
+          }
+          return Promise.reject(new Error("Wrong data"));
+        },
+        (error: unknown) =>
+          genericInternalUnknownErrorHandler(
+            error,
+            "organizationController#createOnboardingDocuments | An error occurred during document generation.",
+            "An error occurred during document generation."
+          )
+      );
+    };
+    const traverseResult = (
```

Comment:
this traverse will be obsolete by the fact we will accept only one request at time